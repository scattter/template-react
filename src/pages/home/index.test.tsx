import { shallow, ShallowWrapper } from 'enzyme';
import { Home } from './index';

describe('test Home page', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Home />);
  });

  it('snapshot test', function () {
    expect(wrapper).toMatchSnapshot();
  });

  it('should success increment count when click button', async () => {
    const strong = wrapper.find('strong').first();
    expect(strong.text()).toEqual('state is 0');
    wrapper.find('button').first().simulate('click');
    await (() => {
      expect(strong.text()).toEqual('state is 1');
    });
  });
});
