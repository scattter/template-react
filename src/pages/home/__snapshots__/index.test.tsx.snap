// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Home page snapshot test 1`] = `
<Fragment>
  <h1>
    This is Home Page
  </h1>
  <strong>
    state is 
    0
  </strong>
  <br />
  <button
    onClick={[Function]}
  >
    revert state
  </button>
</Fragment>
`;
